packages/core/src/components/search/Filter/FilterSlider.tsx

```diff
@@ -111,128 +108,43 @@ function FilterSlider({
-          selected: !!selected.find(({ value }) => value === 'pickup-in-point'),
-          quantity: selectedPickupPoint?.totalItems,
-        }
-      : undefined
-
-  const allDeliveryMethodsFacet = {
-    value: 'all-delivery-methods',
-    label:
-      deliverySettingsData?.deliveryMethods?.allDeliveryMethods ??
-      'All delivery methods',
-    selected: !selected.some(
-      (facet) =>
-        facet.key === 'shipping' && facet.value !== 'all-delivery-methods'
-    ),
-    quantity: 0,
-  }
-
-  let filteredFacets = facets.filter((facet) => facet.key !== 'shipping')
-  if (isDeliveryPromiseEnabled) {
-    filteredFacets = facets.map((facet) => {
-      if (
-        facet.key === 'shipping' &&
-        facet.__typename === 'StoreFacetBoolean'
-      ) {
-        const hasAllDeliveryMethodsFacet = facet.values.some(
-          (item) => item.value === allDeliveryMethodsFacet.value
-        )
-        if (!hasAllDeliveryMethodsFacet) {
-          facet.values = [allDeliveryMethodsFacet, ...facet.values]
-        }
-
-        const pickupInPointFacetIndex = facet.values.findIndex(
-          (item) => item?.value === 'pickup-in-point'
-        )
-
-        // Remove old pickup `pickup in point` facet from list and search state
-        if (pickupInPointFacetIndex !== -1 && !selectedPickupPoint) {
-          if (selected.some(({ key }) => key === 'shipping')) {
-            const selectedShippingFacet = selected.find(
-              ({ key }) => key === 'shipping'
-            )
-            const selectedPickupInPointFacets = selected.filter(
-              ({ key, value }) =>
-                value === 'pickup-in-point' || key === 'pickupPoint'
-            )
-
-            selectedPickupInPointFacets.length !== 0
-              ? togglePickupInPointFacet(selectedPickupInPointFacets)
-              : toggleFilterFacets([selectedShippingFacet])
-          }
-
-          facet.values = facet.values.filter(
-            (_, index) => index !== pickupInPointFacetIndex
-          )
-        }
-        // Prevent multiple `pickup in point` facet
-        else if (pickupInPointFacetIndex === -1 && selectedPickupPoint) {
-          facet.values.push(pickupInPointFacet)
-        }
-        // Replace current `pickup-in-point` facet with the updated one
-        else if (
-          facet.values[pickupInPointFacetIndex] &&
-          facet.values[pickupInPointFacetIndex]?.label !==
-            pickupInPointFacet.label
-        ) {
-          facet.values[pickupInPointFacetIndex] = pickupInPointFacet
-        }
-      }
-
-      return facet
-    })
-  }
+  const {
+    selectedPickupPoint,
+    facets: filteredFacets,
+    deliveryLabel,
+    isPickupAllEnabled,
+    shouldDisplayDeliveryButton,
+    regionalizationData,
+  } = useDeliveryPromise({
+    selectedFacets: selected,
+    toggleFacet: onFacetChange,
+    fallbackToFirst: true,
+    allFacets: facets,
+    deliverySettings: deliverySettings,
+  })
```

Comment:
I believe in this file is missing update the deliverySettings to use the one merged with global settings
```suggestion
  })
  const { deliverySettings: deliverySettingsData } = regionalizationData
```